readthedocs/api/v3/serializers.py

```diff
@@ -1131,6 +1131,25 @@ class Meta:
             "project",
             "_links",
         ]
+        extra_kwargs = {
+            "value": {"write_only": True},
+        }
+
+    def create(self, validated_data):
+        validate_environment_variable_size(
+            project=validated_data["project"],
+            new_env_value=validated_data["value"],
+            error_class=serializers.ValidationError,
+        )
+        return super().create(validated_data)
+
+    def update(self, instance, validated_data):
+        validate_environment_variable_size(
+            project=instance.project,
+            new_env_value=validated_data["value"],
+            error_class=serializers.ValidationError,
+        )
+        return super().update(instance, validated_data)
```

Comment:
Do we need these validations here? Won't this be validated automatically due that you already defined the `clean()` method in the model itself?

Reply:
No, DRF doesn't run the clean method.